src/main/kotlin/com/cjbooms/fabrikt/generators/PropertyUtils.kt

```diff
@@ -68,16 +69,17 @@ object PropertyUtils {
                 ClassType.SUB_MODEL -> {
                     if (this is PropertyInfo.Field && isPolymorphicDiscriminator) {
                         property.addModifiers(KModifier.OVERRIDE)
-                        when (maybeDiscriminator) {
-                            is PropertyInfo.DiscriminatorKey.EnumKey ->
-                                property.initializer("%T.%L", type, maybeDiscriminator.enumKey)
-
-                            is PropertyInfo.DiscriminatorKey.StringKey ->
-                                property.initializer("%S", maybeDiscriminator.stringValue)
-
-                            else -> {
-                                property.addAnnotation(JacksonMetadata.jacksonParameterAnnotation(oasKey))
+                        val discriminators = maybeDiscriminator.getDiscriminatorMappings(modelName)
+                        if (discriminators.size == 1) {
+                            when (val discriminator = discriminators.first()) {
+                                is PropertyInfo.DiscriminatorKey.EnumKey ->
+                                    property.initializer("%T.%L", type, discriminator.enumKey)
+
+                                is PropertyInfo.DiscriminatorKey.StringKey ->
+                                    property.initializer("%S", discriminator.stringValue)
                             }
+                        } else {
+                            property.addAnnotation(JacksonMetadata.jacksonParameterAnnotation(oasKey))
```

Comment:
So this else is for 0 and > 1? Seems a little strange...

Reply:
Yeah, that got me mad, but couldn't find a better way to handle this

That's essentially for covering this case 👉 https://github.com/cjbooms/fabrikt/blob/master/src/test/resources/examples/enumPolymorphicDiscriminator/models/Models.kt#L29

1) If there's no discriminator (size 0) [the property will go to the constructor](https://github.com/cjbooms/fabrikt/blob/master/src/test/resources/examples/enumPolymorphicDiscriminator/models/Models.kt#L13) with the `@param` annotation 
2) If there's only one discriminator (size 1) [the property will be a field with the discriminator value](https://github.com/cjbooms/fabrikt/blob/master/src/test/resources/examples/enumPolymorphicDiscriminator/models/Models.kt#L25) but without the `@param` annotation 
3) But if there are [many discriminators ](https://github.com/cjbooms/fabrikt/blob/master/src/test/resources/examples/enumPolymorphicDiscriminator/api.yaml#L19-L20) for the model class, then the property will have to go to the constructor too without any discriminator value as we can't chose, same as 1)